modules/solid_mechanics/src/materials/IsotropicTempDepHardening.C

```diff
@@ -104,13 +104,15 @@ IsotropicTempDepHardening::initializeHardeningFunctions(unsigned qp)
 
   else if (temp <= _hf_temperatures[0])
   {
-    _hf_index = 0;
+    _hf_index_lo = 0;
+    _hf_index_hi = 1;
```

Comment:
Wouldn't it be better for _hf_index_hi to be 0 in this case?  That would avoid issues if there's just a single hardening function supplied.